libs/deer/src/lib.rs

```diff
@@ -195,152 +195,153 @@ pub trait Visitor<'de>: Sized {
 
-    fn visit_bool(self, v: bool) -> Result<Self::Value, VisitorError> {
+    fn visit_bool(self, value: bool) -> Result<Self::Value, VisitorError> {
         Err(Report::new(TypeError.into_error())
             .attach(ReceivedType::new(bool::document()))
             .attach(ExpectedType::new(self.expecting()))
             .change_context(VisitorError))
     }
 
-    fn visit_number(self, v: Number) -> Result<Self::Value, VisitorError> {
+    fn visit_number(self, value: Number) -> Result<Self::Value, VisitorError> {
         Err(Report::new(TypeError.into_error())
             .attach(ReceivedType::new(Number::reflection()))
             .attach(ExpectedType::new(self.expecting()))
             .change_context(VisitorError))
     }
 
-    fn visit_char(self, v: char) -> Result<Self::Value, VisitorError> {
+    fn visit_char(self, value: char) -> Result<Self::Value, VisitorError> {
         let mut buffer = [0; 4];
-        let v = v.encode_utf8(&mut buffer);
+        let v = value.encode_utf8(&mut buffer);
 
         self.visit_str(v)
             .attach(ReceivedType::new(char::reflection()))
     }
 
-    fn visit_str(self, v: &str) -> Result<Self::Value, VisitorError> {
+    fn visit_str(self, value: &str) -> Result<Self::Value, VisitorError> {
         Err(Report::new(TypeError.into_error())
             .attach(ReceivedType::new(<&str>::reflection()))
             .attach(ExpectedType::new(self.expecting()))
             .change_context(VisitorError))
     }
 
-    fn visit_borrowed_str(self, v: &'de str) -> Result<Self::Value, VisitorError> {
-        self.visit_str(v)
+    fn visit_borrowed_str(self, value: &'de str) -> Result<Self::Value, VisitorError> {
+        self.visit_str(value)
     }
 
-    fn visit_string(self, v: String) -> Result<Self::Value, VisitorError> {
-        self.visit_str(&v)
+    fn visit_string(self, value: String) -> Result<Self::Value, VisitorError> {
+        self.visit_str(&value)
     }
 
-    fn visit_bytes(self, v: &[u8]) -> Result<Self::Value, VisitorError> {
+    fn visit_bytes(self, value: &[u8]) -> Result<Self::Value, VisitorError> {
         Err(Report::new(TypeError.into_error())
             .attach(ReceivedType::new(visitor::BinarySchema::document()))
             .attach(ExpectedType::new(self.expecting()))
             .change_context(VisitorError))
     }
 
-    fn visit_borrowed_bytes(self, v: &'de [u8]) -> Result<Self::Value, VisitorError> {
-        self.visit_bytes(v)
+    fn visit_borrowed_bytes(self, value: &'de [u8]) -> Result<Self::Value, VisitorError> {
+        self.visit_bytes(value)
     }
 
-    fn visit_bytes_buffer(self, v: Vec<u8>) -> Result<Self::Value, VisitorError> {
-        self.visit_bytes(&v)
+    fn visit_bytes_buffer(self, value: Vec<u8>) -> Result<Self::Value, VisitorError> {
+        self.visit_bytes(&value)
     }
 
-    fn visit_array<T>(self, v: T) -> Result<Self::Value, VisitorError>
+    fn visit_array<A>(self, array: A) -> Result<Self::Value, VisitorError>
```

Comment:
small nit: Just wondering about the consistency between the names here. I see most cases use `value` but array and object are `array` and `object` respectively. Any reason for this distinction?

Reply:
Gotchu, that's actually an nice distinction, then